ARProgramming/ARProgramming/ViewModel/Level/LevelDataSource.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  LevelDataSource.swift
+//  ARProgramming
+//  
+//  Created by Emil Nielsen on 09/04/2019.
+//  Copyright © 2019 Emil Nielsen and Kasper Schultz Davidsen. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+class LevelDataSource: NSObject, UICollectionViewDataSource {
+    
+    private var levelsForGrade = [Int: [LevelViewModel]]()
+    private let grade: Int
+    
+    init(grade: Int) {
+        self.grade = grade
+        levelsForGrade[grade] = []
+        
+        let levelGradeConfig = Config.read(configFile: "LevelClasses", toType: LevelGradeConfig.self)!
+        let levelsForGrades = levelGradeConfig.levels
+        
+        for levelName in levelsForGrades[grade-1]{
+            if let level = try? LevelManager.loadLevel(byName: levelName) {
+                levelsForGrade[grade]?.append(LevelViewModel(level: level))
```

Comment:
Why are you using LevelViewModel here and not just Level? Do you need the SCNNode?